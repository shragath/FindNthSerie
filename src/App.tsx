import "./App.css";
import ShowEqForm from "./Components/Form";

function App() {
  return <ShowEqForm />;
}

export default App;
