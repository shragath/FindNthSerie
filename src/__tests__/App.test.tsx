import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import App from '../App';
import renderer from 'react-test-renderer';

// Test renders App components
describe('Renders components sin fallos', () => {
    it('Renders App', () => {
        render(<App />);
    });

    it('Prueba de comparacion snapshot', () => {
        const tree = renderer.create(<App />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
