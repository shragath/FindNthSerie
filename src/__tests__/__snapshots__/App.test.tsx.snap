// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders components sin fallos Prueba de comparacion snapshot 1`] = `
[
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="nInput"
    >
      Escribe el numero (n) de la serie:
    </label>
    <input
      id="nInput"
      placeholder="Number"
      required={true}
      title="n series value"
      type="number"
    />
    <button
      id="buttonCalcular"
      type="submit"
    >
      Calcular
    </button>
  </form>,
  <div>
    <label>
      Equation: serie(
      n
      ) = 2 fibonacci(
      n
       - 2) + 3 triangular(
      n
       - 2) - 7 primo(
      n
       - 1)
    </label>
  </div>,
  <div>
    <p>
      Fibonacci(
      -2
      ) = 
      0
      , Triangular(
      -2
      ) = 
      0
      , Primo(
      -1
      ) =
       
      0
    </p>
  </div>,
  <div>
    <label>
      Resultado = 
      0
       + 
      0
       -
       
      0
       = 
      <b>
        0
      </b>
    </label>
  </div>,
]
`;
